tests/rptest/tests/compatibility/arroyo_test.py

```diff
@@ -31,26 +31,24 @@ def __init__(self, ctx, *args, **kwargs):
     def test_arroyo_test_suite(self):
         test_node = self.preallocated_nodes[0]
 
+        failed = None
         try:
             env_preamble = f"DEFAULT_BROKERS={self.redpanda.brokers()}"
 
-            failed = False
             for line in test_node.account.ssh_capture(
                     f"{env_preamble} "
                     f"python3 -m pytest {ArroyoTest.TEST_SUITE_PATH} "
                     "-k KafkaStreamsTestCase -rf",
                     combine_stderr=True,
                     timeout_sec=120):
                 self.logger.info(line)
-                if 'FAILED' in line:
-                    failed = True
-
-            if failed:
-                assert False, "Arroyo test failures occurred. Please check the log file"
-        except RemoteCommandError as err:
-            if err.exit_status == 2:
-                assert False, "Arroyo test suite was interrupted"
-            elif err.exit_status == 3:
-                assert False, "Internal error during execution of Arroyo test suite"
-            elif err.exit_status == 4:
-                assert False, "Pytest command line invocation error"
+                if 'FAILED' in line or 'FAILURE' in line:
+                    raise Exception("Arroyo test failures occurred. Please check the log file")
+        except Exception as err:
+            failed = err
+        finally:
+            test_node.account.kill_process('arroyo',
+                                           clean_shutdown=False,
+                                           allow_fail=True)
+        if failed is not None:
```

Comment:
You don't have to do this in python finally blocks: if an exception is raised, the finally will still be called before it is bubbled up.  So these two lines can go away and the `except Exception` can too.

Reply:
Good point, fix pushed